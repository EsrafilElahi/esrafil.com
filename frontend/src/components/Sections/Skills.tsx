import { lazy, Suspense } from 'react';
import type { Skill } from '../../../types.ts';
import Fallback from '../Fallback.tsx';

const SkillItem = lazy(() => import('../SkillItem.tsx'));

type Props = {
  skills: Skill[];
};

const Skills = (props: Props) => {
  const { skills } = props;

  return (
    <div className='flex-center flex-wrap gap-10' data-testid='skills'>
      {skills.map((skill) => (
        <Suspense key={skill.id} fallback={<Fallback />}>
          <SkillItem skill={skill} />
        </Suspense>
      ))}
    </div>
  );
};

export default Skills;
