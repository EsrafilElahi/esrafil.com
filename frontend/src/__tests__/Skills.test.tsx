import { act, cleanup, findByTestId, fireEvent, render, screen, waitFor, within } from '@testing-library/react';
import { afterEach, beforeAll, beforeEach, describe, test } from 'vitest';
import Skills from '../components/Sections/Skills';
import mockSkills from '../mocks/skills.json';

beforeAll(() => {
  render(<Skills skills={mockSkills} />);
});

afterEach(() => {
  cleanup();
});

describe('Skills Component', () => {
  test('render Skills', () => {
    const { container } = render(<Skills skills={mockSkills} />);
    expect(container).toBeInTheDocument();
  });

  test('Skills snapshot', () => {
    const result = render(<Skills skills={mockSkills} />);
    expect(result).toMatchSnapshot();
  });

  test('get all skills and test length', async () => {
    await act(async () => {
      render(<Skills skills={mockSkills} />);
    });

    await screen.findAllByText('React').then((items) => {
      expect(items).toHaveLength(9);
    });
  });

  test('all skills has name and icon', async () => {
    await act(async () => {
      render(<Skills skills={mockSkills} />);
    });

    await screen.findAllByText('React').then((items) => {
      for (const item of items) {
        expect(within(item).getByText('React')).toBeInTheDocument();
      }
    });
  });
});
