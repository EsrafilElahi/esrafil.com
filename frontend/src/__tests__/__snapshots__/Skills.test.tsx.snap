// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Skills Component > Skills snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex-center flex-wrap gap-10"
      >
        <div
          class="flex-center gap-2 flex-[0_0_20%] md:flex-[0_0_10%]"
        >
          <img
            alt="React"
            class="w-8 h-8"
            src="/assets/react.svg"
          />
          <span>
            React
          </span>
        </div>
        <div
          class="flex-center gap-2 flex-[0_0_20%] md:flex-[0_0_10%]"
        >
          <img
            alt="React"
            class="w-8 h-8"
            src="/assets/react.svg"
          />
          <span>
            React
          </span>
        </div>
        <div
          class="flex-center gap-2 flex-[0_0_20%] md:flex-[0_0_10%]"
        >
          <img
            alt="React"
            class="w-8 h-8"
            src="/assets/react.svg"
          />
          <span>
            React
          </span>
        </div>
        <div
          class="flex-center gap-2 flex-[0_0_20%] md:flex-[0_0_10%]"
        >
          <img
            alt="React"
            class="w-8 h-8"
            src="/assets/react.svg"
          />
          <span>
            React
          </span>
        </div>
        <div
          class="flex-center gap-2 flex-[0_0_20%] md:flex-[0_0_10%]"
        >
          <img
            alt="React"
            class="w-8 h-8"
            src="/assets/react.svg"
          />
          <span>
            React
          </span>
        </div>
        <div
          class="flex-center gap-2 flex-[0_0_20%] md:flex-[0_0_10%]"
        >
          <img
            alt="React"
            class="w-8 h-8"
            src="/assets/react.svg"
          />
          <span>
            React
          </span>
        </div>
        <div
          class="flex-center gap-2 flex-[0_0_20%] md:flex-[0_0_10%]"
        >
          <img
            alt="React"
            class="w-8 h-8"
            src="/assets/react.svg"
          />
          <span>
            React
          </span>
        </div>
        <div
          class="flex-center gap-2 flex-[0_0_20%] md:flex-[0_0_10%]"
        >
          <img
            alt="React"
            class="w-8 h-8"
            src="/assets/react.svg"
          />
          <span>
            React
          </span>
        </div>
        <div
          class="flex-center gap-2 flex-[0_0_20%] md:flex-[0_0_10%]"
        >
          <img
            alt="React"
            class="w-8 h-8"
            src="/assets/react.svg"
          />
          <span>
            React
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex-center flex-wrap gap-10"
    >
      <div
        class="flex-center gap-2 flex-[0_0_20%] md:flex-[0_0_10%]"
      >
        <img
          alt="React"
          class="w-8 h-8"
          src="/assets/react.svg"
        />
        <span>
          React
        </span>
      </div>
      <div
        class="flex-center gap-2 flex-[0_0_20%] md:flex-[0_0_10%]"
      >
        <img
          alt="React"
          class="w-8 h-8"
          src="/assets/react.svg"
        />
        <span>
          React
        </span>
      </div>
      <div
        class="flex-center gap-2 flex-[0_0_20%] md:flex-[0_0_10%]"
      >
        <img
          alt="React"
          class="w-8 h-8"
          src="/assets/react.svg"
        />
        <span>
          React
        </span>
      </div>
      <div
        class="flex-center gap-2 flex-[0_0_20%] md:flex-[0_0_10%]"
      >
        <img
          alt="React"
          class="w-8 h-8"
          src="/assets/react.svg"
        />
        <span>
          React
        </span>
      </div>
      <div
        class="flex-center gap-2 flex-[0_0_20%] md:flex-[0_0_10%]"
      >
        <img
          alt="React"
          class="w-8 h-8"
          src="/assets/react.svg"
        />
        <span>
          React
        </span>
      </div>
      <div
        class="flex-center gap-2 flex-[0_0_20%] md:flex-[0_0_10%]"
      >
        <img
          alt="React"
          class="w-8 h-8"
          src="/assets/react.svg"
        />
        <span>
          React
        </span>
      </div>
      <div
        class="flex-center gap-2 flex-[0_0_20%] md:flex-[0_0_10%]"
      >
        <img
          alt="React"
          class="w-8 h-8"
          src="/assets/react.svg"
        />
        <span>
          React
        </span>
      </div>
      <div
        class="flex-center gap-2 flex-[0_0_20%] md:flex-[0_0_10%]"
      >
        <img
          alt="React"
          class="w-8 h-8"
          src="/assets/react.svg"
        />
        <span>
          React
        </span>
      </div>
      <div
        class="flex-center gap-2 flex-[0_0_20%] md:flex-[0_0_10%]"
      >
        <img
          alt="React"
          class="w-8 h-8"
          src="/assets/react.svg"
        />
        <span>
          React
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
